controller_manager/test/test_controller_manager.cpp

```diff
@@ -423,23 +423,28 @@ TEST_P(TestControllerUpdateRates, check_the_controller_update_rate)
   EXPECT_EQ(test_controller->get_update_rate(), ctrl_update_rate);
   const auto cm_update_rate = cm_->get_update_rate();
   const auto controller_update_rate = test_controller->get_update_rate();
-  const auto controller_factor = (cm_update_rate / controller_update_rate);
-  const auto expected_controller_update_rate =
-    static_cast<unsigned int>(std::round(cm_update_rate / static_cast<double>(controller_factor)));
 
   const auto initial_counter = test_controller->internal_counter;
-  for (size_t update_counter = 1; update_counter <= 10 * cm_update_rate; ++update_counter)
+  rclcpp::Time time = rclcpp::Time(0);
+  for (size_t update_counter = 0; update_counter <= 10 * cm_update_rate; ++update_counter)
   {
     EXPECT_EQ(
       controller_interface::return_type::OK,
-      cm_->update(rclcpp::Time(0), rclcpp::Duration::from_seconds(0.01)));
+      cm_->update(time, rclcpp::Duration::from_seconds(0.01)));
```

Comment:
can we here have 2 cases of controller. Those that can be perfectly followed and those that can  not?

Reply:
Absolutely, that is already tested using the testingValues, I've added corresponding lines for your reference. 

https://github.com/ros-controls/ros2_control/blob/af1c9a674ed41b7e96814925372ae7058b53da7b/controller_manager/test/test_controller_manager.cpp#L452-L454